apps/extension/src/background/keyring/utils.ts

```diff
@@ -0,0 +1,67 @@
+import { KVStore } from "@anoma/storage";
+
+import { TabStore } from "./types";
+import { ExtensionRequester } from "extension";
+
+/**
+ * Utility to sync tabs in storage with currently available tabs in browser
+ *
+ * @param {TabStore[]} connectedTabsStore
+ * @param {ExtensionRequester} requester
+ * @param {string} chainId
+ *
+ * @return Promise<TabStore[]>
+ */
+export const syncTabs = async (
+  connectedTabsStore: KVStore<TabStore[]>,
+  requester: ExtensionRequester,
+  chainId: string
+): Promise<TabStore[]> => {
+  const tabs = await requester.queryBrowserTabIds();
+  const storedTabs = await connectedTabsStore.get(chainId);
+  const currentTabs: TabStore[] = [];
+
+  if (storedTabs?.length === 0) {
+    return [];
+  }
+
+  storedTabs?.forEach((tab: TabStore) => {
+    if (tabs.indexOf(tab.tabId) > -1) {
```

Comment:
You can also use `includes` . As a side not it is quite funny how long it took to add it to JS  :D

Reply:
ah yes, thanks for the reminder!